app/src/main/java/com/mstudio/superheroarch/presentation/DetailsViewModel.kt

```diff
@@ -23,18 +23,25 @@ class DetailsViewModel(
 
     private suspend fun fetchFirstEpisodeDetails(episodeUrl: String) {
         try {
-            val episodeResult = getEpisodeDetailsUseCase(episodeUrl)
-            episodeResult.getOrNull()?.let {
-                view.displayFirstEpisodeDetails(it)
+            val result = getEpisodeAndDetailsUseCase(episodeUrl)
+            if (result.isSuccess) {
+                val episodeDetailsViewEntity = result.getOrThrow()
+
+                view.displayFirstEpisodeDetails(episodeDetailsViewEntity)
+                view.displayEpisodeRatingAndImage(episodeDetailsViewEntity)
+            } else {
+                view.showError("Failed to load episode details")
             }
         } catch (e: Exception) {
-            view.showError("Failed to load episode details")
+            view.showError("Failed to load episode details: ${e.message}")
         }
     }
+
 }
 
 interface DetailsViewTranslator {
     fun displayCharacterDetails(character: CharacterEntity)
-    fun displayFirstEpisodeDetails(episode: EpisodeEntity)
+    fun displayFirstEpisodeDetails(episodeDetailsViewEntity: EpisodeDetailsViewEntity)
```

Comment:
don't use two different methods with the same param and being called in the same method.
Split them to handle the option of one backend failing and the other working or make the assumption that both are going to work.

Reply:
Solved in next commit